src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -142,6 +142,11 @@ internal static int RouteComparison(RouteEntry x, RouteEntry y)
 
                     if (xSegment.IsParameter)
                     {
+                        if (!xSegment.IsOptional && ySegment.IsOptional)
```

Comment:
Does this need a converse?

```
if (xSegment.IsOptional && !ySegment.IsOptional)
{
   return 1;
}
```

Reply:
Yes, added a comment that we always want to favor non-optional params over optional ones.